lib/assets/build-scripts/junit_grader.json

```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "condition": {
+      "predicate": "dir",
+      "path": "$EXTRACTED/starter"
+    },
+    "on_false": "cp_submission",
+    "on_true": [
+      {
+        "cmd": "cp -r $EXTRACTED/starter/src/. $BUILD/"
+      },
+      {
+        "cmd": "cp -r $EXTRACTED/starter/test/. $BUILD/"
+      }
+    ]
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/submission/src/. $BUILD/",
+    "label": "cp_submission"
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/submission/test/. $BUILD/"
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/junit-jar/* $BUILD/"
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/hamcrest-jar/* $BUILD/"
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/junit-tap-jar/* $BUILD/"
+  },
+  {
+    "cmd": "cp -r $EXTRACTED/annotations-jar/* $BUILD/"
+  },
+  {
+    "condition": {
+      "predicate": "dir",
+      "path": "$EXTRACTED/testing"
+    },
+    "on_true": [
+      {
+        "cmd": "cp -r $EXTRACTED/testing/src/. $BUILD/"
+      },
+      {
+        "cmd": "cp -r $EXTRACTED/testing/test/. $BUILD/"
+      }
+    ]
+  },
+  {
+    "cmd": "find -type f -name '*.java' > compile-list.txt",
```

Comment:
Even this command can be rewritten without needing shell execution.  We can instead write `find -type f -name '*.java' -fprint compile-list.txt`, and then since that command no longer uses shell redirection, we can write it as a normal string[] of command and arguments.